src/queue.ts

```diff
@@ -0,0 +1,719 @@
+  >
+  implements
+    Integration<
+      "Queue",
+      // queue is not directly invokable, only via event bridge pipe.
+      () => any,
+      EventBusTargetIntegration<
+        Message,
+        Omit<aws_events_targets.SqsQueueProps, "message"> | undefined
+      >
+    >
+{
+  /**
+   * @hidden
+   */
+  public static readonly FunctionlessType = "Queue";
+
+  /**
+   * @hidden
+   */
+  readonly functionlessKind = "Queue";
+
+  /**
+   * @hidden
+   */
+  readonly kind = "Queue";
+
+  /**
+   * @hidden
+   */
+  // @ts-ignore - value does not exist, is only available at compile time
+  readonly __functionBrand: () => Promise<void>;
+
+  /**
+   * The ARN of this queue
+   * @attribute
+   */
+  public get queueArn() {
+    return this.resource.queueArn;
+  }
+
+  /**
+   * The Name of this queue
+   * @attribute
+   */
+  public get queueName() {
+    return this.resource.queueName;
+  }
+
+  /**
+   * The URL of this queue
+   * @attribute
+   */
+  public get queueUrl() {
+    return this.resource.queueUrl;
+  }
+
+  constructor(scope: Construct, id: string, props: QueueProps<Message>);
+  constructor(resource: aws_sqs.IQueue, props: QueueProps<Message>);
+  constructor(
+    ...args:
+      | [secret: aws_sqs.IQueue, props: QueueProps<Message>]
+      | [scope: Construct, id: string, props: QueueProps<Message>]
+  ) {
+    // @ts-ignore
+    super(...args);
+
+    const queueUrl = this.queueUrl;
+    const codec = (this.props.serializer ?? Serializer.json())?.create();
+    const lambdaQueueUrlRetriever = this.props.lambda?.queueUrlRetriever;
+
+    function serialize(message: Message): string {
+      if (codec) {
+        return codec.write(message);
+      } else if (typeof message === "string") {
+        return message;
+      } else if (Buffer.isBuffer(message)) {
+        return message.toString("utf8");
+      } else {
+        throw new Error(
+          `Message must be a string if there is no 'serializer' configured, ${message}`
+        );
+      }
+    }
+
+    this.sendMessage = makeIntegration<
+      "AWS.SQS.SendMessage",
+      (input: SendMessageRequest<Message>) => Promise<AWS.SQS.SendMessageResult>
+    >({
+      kind: "AWS.SQS.SendMessage",
+      native: {
+        bind: (func) => this.resource.grantSendMessages(func.resource),
+        preWarm: (context) => context.getOrInit(SQSClient),
+        call: async ([input], context) => {
+          const sqs = context.getOrInit(SQSClient);
+
+          const messageBody = serialize(input.Message);
+
+          // @ts-ignore
+          delete input.Message;
+
+          const updatedQueueUrl =
+            lambdaQueueUrlRetriever?.(queueUrl) ?? queueUrl;
+
+          const response = await sqs
+            .sendMessage({
+              ...input,
+              MessageBody: messageBody,
+              QueueUrl: updatedQueueUrl,
+            })
+            .promise();
+
+          return response;
+        },
+      },
+    });
```

Comment:
whats the plan to support SFN?

Reply:
I need to implement it. Forgot twice lol

Reply:
Implemented sendMessage, sendMessageBatch, purge and receiveMessage